cmd/run.go

```diff
@@ -455,9 +456,22 @@ a commandline interface for interacting with it.`,
 				Metrics: engine.Metrics,
 				Time:    engine.Executor.GetTime(),
 			}
-			// TODO: add json summary later
-			switch {
-			default:
+			if conf.ReportSummary.Valid {
```

Comment:
We won't get to this part if `NoSummary` is set which is IMO not what we want ... 
Maybe the options are named badly but I think this should work as follows:
`NoSummary` means "don't print (the current) summary to the stdout", `ReportSummary` means `write a json representation of the summary to a file` . Both can be set independantly so that you both write the json to a file and print the current summary to the stdout, neither or only one of those ...  I find this to be the approriate behaviour 
cc @na-- @imiric

Reply:
hmm not totally sure, but it makes sense, and would be more flexible, so a tentative :+1: (don't have a lot of time to investigate implications)

just keep in mind that the current `NoSummary` option is used (together with `NoThresholds`) to determine if we can spare some calculation work: https://github.com/loadimpact/k6/blob/f09a8e8b3677bcf707bba77ae45fadac069dc927/core/engine.go#L395-L397

so, if enabling `SummaryReport` could independently require k6 to calculate the stats even if `NoSummary` is specified, these checks would have to also include `SummaryReport`

and we'll need to specify in the CLI and other docs that `NoSummary` refers only to the human readable stdout report

Reply:
Yeah I though of this when I first saw the above code, but ... obviously forgot by the time I wrote the comment :(